docs/cookie_rotation.md

```diff
@@ -0,0 +1,82 @@
+## Session cookie rotation
+
+### About
+The session cookie, amongst others, is encrypted based on the value of the `secret_key_base` secret. If this
+secret changes then all existing session cookies become invalid and users would be forced to login again or
+encounter errors. To rotate the `secret_key_base` secret, for security reasons, while mitigating the impact
+on users you can follow this guide.
+
+### Guide
+Encrypted cookie rotation is performed by the `config/initializers/cookie_rotation.rb` initializer
+but it will only take action if there is an `old_secret_key_base` secret in `config/secrets.yml`.
+
+To rotate the `secret_key_base` without inconveniencing users you must:
+
+1. add an old_secret_key_base to `config/secrets.yml`
+   and assign its value to that of the current secret
+   you wish to rotate.
+
+2. generate a new secret at a terminal rooted in the
+   (or any rails) repo using `rails secret` and assign
+   the existing `secret_key_base`'s value to be that
+   of the new secret
+   example:
+   ```
+   # config/secrets.yml
+   development:
+     old_secret_key_base: my-current-local-secret-key-base-secret
+     secret_key_base: my-new-local-secret-key-base-secret
+   ....
+   production:
+     old_secret_key_base: ENV["OLD_SECRET_KEY_BASE"]
+     secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
+   ```
+
+3. For production, as in the above example, you will need to
+   add an OLD_SECRET_KEY_BASE env variable to each namespace/host,
+   add the current secret to that and modify the existing
+   SECRET_KEY_BASE to hold the new secret.
+
+   Note: secret env vars are kept in `kubernetes_deploy/<environment>/secrets.yaml`
+   , where `environment` can be dev, staging, production or
+   any other "namespace".
+
+4. Deploy the application, ensuring the new secrets
+   are applied first. The `kubernetes_deploy/scripts/deploy.sh`
+   or `.circleci/deploy.sh` scripts will apply the secrets
+   before the image, thereby ensuring this.
+
+5. Leave the rotation in place for x, where x is however
+   long (days?!) we think it will take the majority of
```

Comment:
Do users get automatically logged out if they haven't access the site in a certain period of time? If so, that could be you *x* days.

Reply:
Essentially yes,

if user `A` has visited site during the x days the old/new secret rotation config is active, but user `B` has not then when we remove the old_secret_key_base value (ENV var on hosted environments) after the x days:

1. user  `A` would not encounter any problem
2. user `B` would be logged out/suffer i.e. encounter `InvalidAuthenticityToken` error